snakemake_rules/Snakefile

```diff
@@ -0,0 +1,134 @@
+        expand(COMMON_DIR + "/{sample}/multiqc/report.html", sample=config["read1"]),
+
+rule fastqc:
+    input:
+        read1 = lambda wildcards: config["read1"][wildcards.sample],
+        read2 = lambda wildcards: config["read2"][wildcards.sample]
+    output:
+        qc_read1 = COMMON_DIR + "/{sample}/fastqc/{sample}_1_fastqc.zip",
+        qc_read2 = COMMON_DIR + "/{sample}/fastqc/{sample}_2_fastqc.zip"
+    params:
+        tmp_dir = config["tmp_dir"],
+        outpath = COMMON_DIR + "/{sample}/fastqc/"
+    threads: 15
+    shell:
+        "fastqc --threads {threads} --noextract --dir {params.tmp_dir} --outdir {params.outpath} {input.read1};"
+        "fastqc --threads {threads} --noextract --dir {params.tmp_dir} --outdir {params.outpath} {input.read2};"
+
+rule trim_galore_pe:
+    input:
+        read1 = lambda wildcards: config["read1"][wildcards.sample],
+        read2 = lambda wildcards: config["read2"][wildcards.sample]
+    output:
+        trimmed_1 = COMMON_DIR + "/{sample}/trim-galore/{sample}_1_val_1.fq.gz",
+        trimmed_2 = COMMON_DIR + "/{sample}/trim-galore/{sample}_2_val_2.fq.gz"
+    params:
+        outpath = COMMON_DIR + "/{sample}/trim-galore/"
+    shell:
+         "trim_galore --cores 15 --fastqc --gzip --output_dir {params.outpath} --paired {input.read1} {input.read2}"
+
+rule star_index:
+    input:
+         fa = config["genome"],
+         gtf = config["gtf"]
+    output:
+         directory(COMMON_DIR + "/star_index/")
+    threads: 36
+    shell:
+         'STAR --runThreadN {threads} '
+         '--runMode genomeGenerate '
+         '--genomeDir {output}'
+         '--genomeFastaFiles {input.fa} '
+         '--sjdbGTFfile {input.gtf} '
+         '--sjdbOverhang 99'
+
+rule star_align:
+    input:
+        trimmed_1 = COMMON_DIR + "/{sample}/trim-galore/{sample}_1_val_1.fq.gz",
+        trimmed_2 = COMMON_DIR + "/{sample}/trim-galore/{sample}_2_val_2.fq.gz",
+        ref_dir = COMMON_DIR + "/star_index/"
+    output:
+        COMMON_DIR + "/{sample}/star_aln/{sample}.Aligned.sortedByCoord.out.bam",
+        COMMON_DIR + "/{sample}/star_aln/{sample}.Aligned.toTranscriptome.out.bam"
+    params:
+        prefix = COMMON_DIR + "/{sample}/star_aln/{sample}."
+    log:
+        COMMON_DIR + "/{sample}/star_aln/logs/{sample}.log",
+    threads: 36
+    shell:
+        "STAR --genomeDIR {input.ref_dir} "
+        "--readFilesin {input.trimmed_1} {input.trimmed_2} "
+        "--readFilesCommand zcat "        
+        "--twopassMode Basic "
+        "--quantMode TranscriptomeSAM " # using RSEM
+        "--peOverlapNbasesMin 10 "
+        "--outSAMtype BAM SortedByCoordinate "
+        "--outFileNamePrefix {params.prefix} "
+        "--runThreadN {threads}"
+
+
+# rule rename_bam: # I will add this rule .. later to comply w/ conventions ^_^
+
+rule mark_dupes:
+    input:
+        COMMON_DIR + "/{sample}/star_aln/{sample}.Aligned.sortedByCoord.out.bam"
```

Comment:
@jemten which BAM file should I be using here? 

1. `.Aligned.sortedByCoord.out.bam"`
2. `.Aligned.toTranscriptome.out.bam"`

Reply:
I've only ever produced the Aligned.sortedByCoord.out.bam with star. I would say that that file needs to go through mark duplicates. I don't know about Aligned.toTranscriptome.out.bam. Might be that it's needed. I actually think the bam file needs to be sorted by coordinates before markduplicates. So then you would need to sort that file first. For that you can use the `samtools sort` command. Just be sure to use it with the `-T` option and point it to your local temp folder (`~/tmp`). I forgot that once and filled up the "global" /tmp, causing slurm to crash 😬